### PR TITLE
docs(nextjs13) updated documentation for stable app dir and server components

### PR DESCRIPTION
**Pull request by Hamzah-syed**
## What kind of change does this PR introduce?

This pull request introduces an update regarding the stable status of the `app` directory and Server Components in Next.js 13.4.

## What is the current behavior?

Currently, Next.js 13 is stable, but the `app` directory and Server Components are still in beta.

## What is the new behavior?

With this update, I've confirmed that both the `app` directory and Server Components have reached a stable status in Next.js 13.4.

## Additional context

This change is important as it ensures that developers are aware of the stable nature of the app directory and Server Components in Next.js 13. It eliminates any confusion and provides clarity for users of these features.

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx [hunk 1]
```diff
@@ -28,7 +28,7 @@ This library supports the following tooling versions:
 - Node.js: `^10.13.0 || >=12.0.0`
 - Next.js: `>=10`
 
-> Note: Next.js 13 is stable, however, the new `app` directory and Server Components are still in beta. Check out our experimental guide on [using Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
+> Note: In [Next.js 13.4](https://nextjs.org/blog/next-13-4), The `app` directory and Server Components have reached stable status. Therefore, the new features are no longer in beta. You can refer to our experimental guide on utilizing [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
```
Comment (gregnr) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx:31]:
```suggestion
> Note: As of [Next.js 13.4](https://nextjs.org/blog/next-13-4), the `app` directory and Server Components have reached stable status. Check out our guide on using [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
```

Comment (gregnr) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx:31]:
```suggestion
> Note: As of [Next.js 13.4](https://nextjs.org/blog/next-13-4), the `app` directory and Server Components have reached stable status. Check out our guide on using [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
```

**Review by gregnr — COMMENTED** [review]
Thanks @Hamzah-syed! Do you mind updating the nextjs-server-components page to remove the beta comment?

https://github.com/supabase/supabase/blob/c8501f19d21b0031638bdad4d7450b731e841ad0/apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx#L13-L14

**Comment by Hamzah-syed** [comment]
sure @gregnr I'm on it.

**Comment by Hamzah-syed** [comment]
Done @gregnr please review it.

**Review by gregnr — APPROVED** [review]
Excellent, thank you @Hamzah-syed!

**Pull request merged**